### PR TITLE
🔍 Added Bing Search Plugin

### PR DESCRIPTION
**Pull request by itzraiss**
## Summary

Added the Bing Search plugin, now enabling users to utilize both Bing and Google for conducting internet searches on both the Plugins and Assistants endpoints. Additionally, instructions have been added to the documentation for obtaining the responsible API.

## Change Type

Please delete any irrelevant options.

- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [x] Documentation update

## Checklist

- [x] My code adheres to this project's style guidelines
- [ ] I have performed a self-review of my own code
- [x] I have made pertinent documentation changes
- [x] I have written tests demonstrating that my changes are effective or that my feature works
- [x] New documents have been locally validated with mkdocs

Plugin:

https://github.com/danny-avila/LibreChat/assets/140329135/170cd224-ca50-41a4-b522-0360cadf2189

**Asssitants:**
image

**Plugin:**
image

### CODE REVIEW

**Comment by itzraiss** [comment]
I still lack experience in programming; much of the code I've used as a basis is derived from existing sources (GoogleSearch). If the pull request is deemed relevant and useful for implementation in LibreChat, I'd appreciate assistance in rectifying these errors. 😅

**Comment by kneelesh48** [comment]
This is good keep going

File: api/app/clients/tools/structured/BingSearch.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        entities,
+        news,
+        computation,
+        timeZone,
+        translations,
+        spellSuggestion,
+        rankingResponse,
+      };
+
+      return JSON.stringify(results);
```
Comment (cpbotha) [inline api/app/clients/tools/structured/BingSearch.js:87]: The webPages object is a list of search hits with "snippets", really short extracts of the found pages [1].

It looks to me like this code will only pass back these snippets to the OpenAI Assistants API, and not the full page contents, or did I misunderstand?

Usually with the previous API, you would have to scrape each of the pages and extract important contents with readability.js before passing that back as the response.

[1] https://learn.microsoft.com/en-us/bing/search-apis/bing-web-search/search-responses#webpages-answer

**Comment by cpbotha** [comment]
This tool would be really great. Azure Search has a free tier which would work well for many users.

I am not the reviewer, just an external interested party, and have left a single comment (so far).

**Pull request closed without merging**